juju/framework.py

```diff
@@ -391,7 +405,7 @@ def load_snapshot(self, handle):
         raw_data = self._storage.load_snapshot(handle.path)
         if not raw_data:
             raise NoSnapshotError(handle.path)
-        data = pickle.loads(raw_data)
+        data = pickle.loads(bytes(raw_data))
```

Comment:
When you put bytes into sqlite3, what you get back should be bytes. If so, this would be just copying the memory without a real benefit.